text-generation-inference/docker/Dockerfile

```diff
@@ -55,14 +62,29 @@ RUN apt-get update -y \
     python3-pip \
     python3-setuptools \
     python-is-python3 \
+    git \
     && rm -rf /var/lib/apt/lists/* \
     && apt-get clean
 RUN pip3 --no-cache-dir install --upgrade pip
 
-# VERSION is a mandatory parameter
-ARG VERSION
+ARG ENABLE_GCP_INTEGRATION
+ARG VERSION='0.2.2'
 RUN test -n ${VERSION:?}
 
+COPY . /opt/optimum-tpu
+
+RUN if [ -n "$ENABLE_GCP_INTEGRATION" ]; then \
+    # If we are building for GCP, we need to clone the optimum-tpu repo as this is built from the huggingface/Google-Cloud-Containers repository and not the huggingface/optimum-tpu repository
+        rm -rf /opt/optimum-tpu && \
+        git clone https://github.com/huggingface/optimum-tpu.git /opt/optimum-tpu && \
+        cd /opt/optimum-tpu && git checkout v${VERSION}; \
+    fi && \
+    # Check if the optimum-tpu repo is cloned properly
+    if [ ! -d "/opt/optimum-tpu/optimum" ]; then \
+        echo "Error: You are trying to build the image not from optimum-tpu repo. If you are building it from google-cloud-containers repo, please set ENABLE_GCP_INTEGRATION=1 to clone optimum-tpu repo automatically" && exit 1; \
+    fi
```

Comment:
I think this is confusing and error-prone. If I get it right you want to use the same Dockerfile, in two different locations. 
You copy the sources to the destination, then you delete them. I would avoid this, just copy only when the integration is not available. We could even consider adding a script in the Google Cloud Containers repo to clone or download optimum-tpu, so you would be sure you are using a fixed version.